src/keydra/providers/splunk.py

```diff
@@ -57,44 +125,36 @@ def _rotate_secret(self, secret):
         :returns: New secret ready to distribute
         :rtype: :class:`dict`
         '''
-        username = self._credentials['key']
-        current_passwd = self._credentials['secret']
 
         # Generate new random password from SecretsManager
         new_passwd = self._generate_splunk_passwd(32)
 
         # Change password on each Splunk host listed in the secret
-        for host in secret['hosts']:
-            try:
-                sp_client = SplunkClient(
-                    username=username,
-                    password=current_passwd,
+        for host in secret['config']['hosts']:
+            secrettype = secret['config'].get('type', 'user')
+
+            if secrettype == 'user':
+                key, password = self._rotate_user(
+                    secret,
                     host=host,
-                    verify=self._verify
-                )
-                sp_client.change_passwd(
-                    username=username,
-                    oldpasswd=current_passwd,
                     newpasswd=new_passwd
                 )
-            except Exception as e:
-                raise RotationException(
-                    'Error rotating user {} on Splunk host '
-                    '{} - {}'.format(username, host, e)
-                )
+
+            if secrettype == 'hectoken':
+                key, password = self._rotate_hec_secret(secret, host)
 
             LOGGER.info(
-                'Successfully changed Splunk user {} ({}) on server {}'.format(
-                    username,
-                    secret['key'],
+                'Successfully changed Splunk {} {} on server {}'.format(
+                    secrettype,
+                    key,
                     host
                 )
             )
 
         return {
```

Comment:
Hmm we may want to rethink the support for multiple hosts in a single secret. Suppose we have two hosts, A and B:
- When `secrettype == 'user'`, suppose rotation on host A succeeds, but fails on host B. This will raise an error on line 73, and (I think) the secret won't be distributed. Subsequent rotation will fail a the password has been updated on host A, but hasn't been distributed to Secrets Manager.
- When `secrettype == 'hectoken'`... UPD: after reading the validation code, I see we only support a single host